ui/admin/app/components/form/policy/policy-selection/index.js

```diff
@@ -0,0 +1,99 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class FormPolicySelectionComponent extends Component {
+  //attributes
+  @tracked isCustomRetentionSelected = this.args.model.retain_for?.days > 0;
+  @tracked isCustomDeletionSelected = this.args.model.delete_after?.days > 0;
+  @tracked retainForOverridable = this.args.model.retain_for?.overridable;
+  @tracked deleteAfterOverridable = this.args.model.delete_after?.overridable;
+
+  //methods
+  /**
+   * This is used to show/hide the custom input field
+   * @type {boolean}
+   */
+  get showCustomInput() {
+    if (this.args.name === 'retention_policy') {
+      return this.isCustomRetentionSelected;
+    } else if (
+      this.args.name === 'deletion_policy' &&
+      this.args.model.retain_for?.days > 0
+    ) {
+      return this.isCustomDeletionSelected;
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Returns true if the toggle is on
+   * @type {boolean}
+   */
+  get isOverridable() {
+    return this.args.model[this.args.customInputName]?.overridable;
+  }
+  //actions
+  /**
+   * Handles custom input changes
+   */
+  @action
+  handleInputChange({ target: { value, name: field } }) {
+    this.args.model[field] = {
+      ...this.args.model[field],
+      days: value || null,
+    };
+  }
+
+  /**
+   * Toggle retain for & delete after overridables
+   */
+  @action
+  handleOverridableToggle({ target: { name: field } }) {
+    this.args.model[field] = {
+      ...this.args.model[field],
+      overridable: !this.args.model[field].overridable,
+    };
+  }
+
+  /**
+   * Show custom text field when custom option is selected
+   */
+  @action
+  handlePolicyTypeSelection({ target: { value: selectedVal, name: policy } }) {
```

Comment:
I notice when I save the default dropdowns without touching on a new create, I get null
<img width="466" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/5783847/835a1c63-8902-4419-a5dd-167d5ca31a04">

but patching the exact same thing without touching it gets me an empty object:
<img width="412" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/5783847/3cb4d69e-0ed6-40a3-a222-f6690b4fc709">

and if I change the dropdown and then back I get the strings `"0"`
<img width="446" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/5783847/8edae9c0-a16a-4681-861d-0a661a6b0b65">

and yet the forever option is the integer `0` and still string `-1` for do not delete
<img width="285" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/5783847/beafb05e-bfa5-42ae-b21f-1696894c8c79">

What are the correct values we should be sending?

Reply:
Interestingly, the API doesn't throw any errors.. but the value they return is in `int` and no object if we don't send a value or an empty obj or null. I'll update to be consistent